src/codegen.cpp

```diff
@@ -10235,8 +10239,14 @@ extern "C" void jl_init_llvm(void)
     const char *jit_profiling = getenv("ENABLE_JITPROFILING");
 
 #if defined(JL_USE_INTEL_JITEVENTS)
-    if (jit_profiling && atoi(jit_profiling)) {
-        jl_using_intel_jitevents = 1;
+    if (jit_profiling) {
+        if (atoi(jit_profiling)) {
+            jl_using_intel_jitevents = 1;
+        }
+    } else {
+#ifdef USE_ITTAPI
+        jl_using_intel_jitevents = __itt_get_collection_state() == __itt_collection_init_successful;
```

Comment:
ittapi itself also checks for `__itt_collection_collector_exists`: https://github.com/intel/ittapi/blob/8e526a82cc805665ef4513399aaea8a5dccfd66f/src/ittnotify/ittnotify_static.c#L1459-L1472
Not clear to my what the exact semantics are (maybe for delayed profiling?), but it seems reasonable to also enable on that state.

Reply:
There is also https://github.com/intel/ittapi/blob/8e526a82cc805665ef4513399aaea8a5dccfd66f/src/ittnotify/jitprofiling.c#L107